openstack/baremetalinspector/v1/introspection/requests.go

```diff
@@ -0,0 +1,115 @@
+package introspection
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// ListIntrospectionsOptsBuilder allows extensions to add additional parameters to the
+// ListIntrospections request.
+type ListIntrospectionsOptsBuilder interface {
+	ToIntrospectionsListQuery() (string, error)
+}
+
+// ListIntrospectionsOpts allows the filtering and sorting of paginated collections through
+// the Introspection API. Filtering is achieved by passing in struct field values that map to
+// the node attributes you want to see returned. Marker and Limit are used
+// for pagination.
+type ListIntrospectionsOpts struct {
+	// Requests a page size of items.
+	Limit int `q:"limit"`
+
+	// The ID of the last-seen item.
+	Marker string `q:"marker"`
+}
+
+// ToIntrospectionsListQuery formats a ListIntrospectionsOpts into a query string.
+func (opts ListIntrospectionsOpts) ToIntrospectionsListQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	return q.String(), err
+}
+
+// ListIntrospections makes a request against the Inspector API to list the current introspections.
+func ListIntrospections(client *gophercloud.ServiceClient, opts ListIntrospectionsOptsBuilder) pagination.Pager {
+	url := listIntrospectionsURL(client)
+	if opts != nil {
+		query, err := opts.ToIntrospectionsListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+	return pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
+		var rpage = IntrospectionPage{pagination.LinkedPageBase{PageResult: r}}
+		return rpage
+	})
+}
+
+// GetIntrospectionStatus makes a request against the Inspector API to get the
+// status of a single introspection.
+func GetIntrospectionStatus(client *gophercloud.ServiceClient, NodeID string) (r GetIntrospectionStatusResult) {
+	_, r.Err = client.Get(introspectionURL(client, NodeID), &r.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{200},
+	})
+	return
+}
+
+// StartOptsBuilder ...
+type StartOptsBuilder interface {
+	ToStartIntrospectionQuery() (string, error)
+}
+
+// StartOpts ...
+type StartOpts struct {
+	// Whether the current installation of ironic-inspector can manage PXE booting of nodes.
+	ManageBoot bool `q:"manage_boot"`
```

Comment:
What's going to be send by default? The correct answers are "nothing" or `true`, `false` is a specific mode.

Reply:
for `q` tags (these are internal to Gophercloud), zero-values are not sent. So if `ManageBoot` isn't specified, then `?manage_boot=false` won't be sent in the request.

If sending false is a valid thing, then this will need changed to `*bool`.

Reply:
Sending `manage_boot=false` is a valid case indeed.

Reply:
changed